tests/Extensions/Hosting.Services.UnitTests/HostBuilderExtensionsTests.cs

```diff
@@ -0,0 +1,241 @@
+		[DataRow(typeof(IServiceContextAccessor<ServiceContext>), typeof(ServiceContextAccessor<StatefulServiceContext>))]
+		public void CheckBuildeStatefulService(Type type, Type? expectedImplementationType)
+		{
+			object obj = new HostBuilder()
+				.BuildStatefulService("TestStatefulService", c => { })
+				.Services
+				.GetService(type);
+
+			Assert.IsNotNull(obj);
+
+			if (expectedImplementationType != null)
+			{
+				Assert.IsInstanceOfType(obj, expectedImplementationType);
+			}
+		}
+
+
+		[TestMethod]
+		public void TestExceptionHandlingAndReportingFroBuildStatelessService()
+		{
+			string serviceType = "StatelessFailedServiceName";
+			CustomEventListener listener = new CustomEventListener();
+			listener.EnableEvents(ServiceInitializationEventSource.Instance, EventLevel.Error);
+
+			Assert.ThrowsException<AggregateException>(() =>  new HostBuilder()
+				.ConfigureServices(c => c.AddTransient<TypeThatShouldNotBeResolvable>())
+				.BuildStatelessService(serviceType, c => { }),
+				"BuildStatelessService should fail in case of unresolvable dependencies");
+
+			bool hasErrorEvent = listener.EventsInformation.Any(e =>
+				serviceType == GetPayloadValue<string>(e, ServiceTypePayloadName)
+				&& e.EventId == (int)EventSourcesEventIds.ServiceHostInitializationFailedEventId);
+
+			Assert.IsTrue(hasErrorEvent, "BuildStatelessService error should be logged");
+		}
+
+
+		[TestMethod]
+		public void TestExceptionHandlingAndReportingForBuildStatefulService()
+		{
+			string serviceType = "StatefulFailedServiceName";
+			CustomEventListener listener = new CustomEventListener();
+			listener.EnableEvents(ServiceInitializationEventSource.Instance, EventLevel.Error);
+
+			Assert.ThrowsException<AggregateException>(() => new HostBuilder()
+				.ConfigureServices(c => c.AddTransient<TypeThatShouldNotBeResolvable>())
+				.BuildStatefulService(serviceType, c => { }),
+				"BuildStatefulService should fail in case of unresolvable dependencies");
+
+			bool hasErrorEvent = listener.EventsInformation.Any(e =>
+				serviceType == GetPayloadValue<string>(e, ServiceTypePayloadName)
+				&& e.EventId == (int)EventSourcesEventIds.ServiceHostInitializationFailedEventId);
+
+			Assert.IsTrue(hasErrorEvent, "BuildStatefulService error should be logged");
+		}
+
+
+		public class TypeThatShouldNotBeResolvable
+		{
+			public TypeThatShouldNotBeResolvable(TypeThatIsNotRegistred value) { }
+
+			public class TypeThatIsNotRegistred { }
+		}
+
+
+		private class CustomEventListener : EventListener
+		{
+			public List<EventWrittenEventArgs> EventsInformation { get; } = new List<EventWrittenEventArgs>();
+
+
+			protected override void OnEventSourceCreated(EventSource eventSource)
+			{
+				base.OnEventSourceCreated(eventSource);
+			}
+
+
+			protected override void OnEventWritten(EventWrittenEventArgs eventData) => EventsInformation.Add(eventData);
+		}
+
+
+		private static T? GetPayloadValue<T>(EventWrittenEventArgs info, string name)
```

Comment:
Use a better name

Reply:
I presume you are meaning generic parameter, that one renamed